cfme/tests/cloud/test_providers.py

```diff
@@ -353,6 +354,68 @@ def test_api_port_max_character_validation_cloud(appliance):
         assert text == prov.default_endpoint.api_port[0:15]
 
 
+@pytest.mark.tier(2)
+@pytest.mark.uncollectif(lambda provider: not provider.one_of(AzureProvider))
+def test_azure_subscription_required(request, provider):
+    """
+    Tests that provider can't be added w/o subscription
+
+    Metadata:
+        test_flag: crud
+    """
+    provider.subscription_id = ''
+    request.addfinalizer(provider.delete_if_exists)
+    flash = ('Credential validation was not successful: '
+            'Incorrect credentials - check your Azure Subscription ID')
+    with pytest.raises(AssertionError, match=flash):
+        provider.create()
+
+
+@pytest.mark.tier(2)
+@pytest.mark.usefixtures('has_no_cloud_providers')
+def test_azure_multiple_subscription(appliance, request, soft_assert):
+    """
+    Verifies that different azure providers have different resources access
+
+    Steps:
+    1. Add all Azure providers
+    2. Compare their VMs/Templates
+
+    Metadata:
+        test_flag: crud
+    """
+    providers = conf.cfme_data.get("management_systems", {})
+    azure_providers = []
+    for provider in providers.items():
+        if provider[1].get('type') == 'azure':
+            azure_providers.append(provider)
+    prov_inventory = []
+    for provider in azure_providers:
```

Comment:
**REQUIRED:** There is no protection here against there being only 1 azure provider.